src/ViewHelpers/BiscuitViewHelper.php

```diff
@@ -12,21 +10,46 @@ class BiscuitViewHelper
      * @param array $biscuits
      * @return string
      */
-    public static function displayAllBiscuits(array $biscuits): string
+    public static function displayAllBiscuits(array $biscuits, array $dunkFlunkData): string
     {
         $result = "";
 
         foreach($biscuits as $biscuit)
         if ($biscuit instanceof Biscuit) {
-            $result .= '<div class="card d-flex flex-direction-column align-items-center p-3 m-4 col-10 col-lg-3">';
+            $dunkFlunkState = null;
+            if(array_key_exists($biscuit->getId(), $dunkFlunkData)) {
+                $dunkFlunkState = $dunkFlunkData[$biscuit->getId()];
+            }
+            $result .= '<div id="' . $biscuit->getId() . '" class="card d-flex flex-direction-column align-items-center p-3 m-4 col-10 col-lg-3">';
             $result .= '<div class="card-title card-background rounded">';
             $result .= '<h2 class="text-center p-2">' . $biscuit->getName() . '</h2>';
             $result .= '</div><div class="card-img d-flex justify-content-center mb-3">';
             $result .= '<img src="' . $biscuit->getImg() . '" class="rounded mw-100" alt="' . $biscuit->getName() . '" />';
             $result .= '</div><form action="biscuitdetails.php" method="GET">';
-            $result .= '<input name="id" type="hidden" value="' . $biscuit->getId() . '" />';
+            $result .= '<input type="hidden" name="id" value="' . $biscuit->getId() . '" />';
             $result .= '<button type="submit" class="btn btn-light">More Info</button>';
-            $result .= '</form></div>';
+            $result .= '</form>';
+            if(isset($dunkFlunkState)) {
+                $result .= '<div class="card-background rounded text-center m-1 fw-bold"><p class="my-auto">';
+                $result .= ($dunkFlunkState ? 'You dunked that biscuit!' : 'You flunked that biscuit!');
```

Comment:
slight copy change suggestion for clarity - "You have dunked this biscuit!" / "You have flunked this biscuit!"